### DISCUSSION TITLE
Loss functions used in the InternLM-XComposer2.5 & its working

### DISCUSSION THREAD

**Discussion by khyati2396** [InternLM/InternLM-XComposer#420] [General]
Hello, InterLM community,

I am getting straight to the point,
**_I need to know what all loss functions are being used in the InternLM-XComposer2.5 and how the loss functions work for this perticular VLM._**

I have already read the previous interLM-XComposer papers but could not find any explanation of the loss functions. 
Below are the documents that I have gone through, 

- interlLM2 
- internLM-XComposer
- internLM-XComposer-2
- internLM-XComposer-4KHD
- internLM-XComposer-2.5

As far as I know 

> **internLM uses:**: original ranking loss function inspired by Focal Loss, which has a difficulty coefficient to the ranking loss.
> **ViT uses** :: Symmetric cross-entropy loss function

I also found this piece of line in the internLM-XComposer2 paper, 
> It is pretrained in an image-language contrastive manner(CLIP)

**_Does it follow a similar learning technique in internLM-XComposer2.5?_**

I must admit this piece of research is a gem for people who need strong VLMs.
It just needs more information related to the loss functions and the model's in-depth training.

Any and all responses are welcomed.

Thanks in advance,
Khyati
